charmhelpers/core/hookenv.py

```diff
@@ -357,18 +358,16 @@ def _implicit_save(self):
 @cached
 def config(scope=None):
     """Juju charm configuration"""
-    config_cmd_line = ['config-get']
-    if scope is not None:
-        config_cmd_line.append(scope)
-    else:
-        config_cmd_line.append('--all')
-    config_cmd_line.append('--format=json')
+    global cache_config
+    config_cmd_line = ['config-get', '--all', '--format=json']
     try:
-        config_data = json.loads(
-            subprocess.check_output(config_cmd_line).decode('UTF-8'))
+        if not cache_config:
```

Comment:
This needs to be `if cache_config is None:`

This stops the false/true check for "{}" and makes it explicit that this will only happen once (unless json.loads fails, that is)